### PR TITLE
feat: write tests for AWS SES

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Follow up of https://github.com/cds-snc/notification-api/pull/1260

Refactor + writing tests for 2 main cases:
- TXT + HTML, without attachments
- TXT + HTML + attachments

### CODE REVIEW

File: tests/app/clients/test_aws_ses.py [hunk 1]
```diff
@@ -80,6 +82,111 @@ def test_send_email_handles_reply_to_address(notify_api, mocker, reply_to_addres
...
+            body='email body',
+            html_body='<p>email body</p>',
+            reply_to_address='reply@example.com',
+        )
+
+    boto_mock.send_raw_email.assert_called_once()
+    raw_message = boto_mock.send_raw_email.call_args.kwargs['RawMessage']['Data']
+
+    regex = dedent(r"""
+        Content-Type: multipart\/alternative; boundary="===============(?P<boundary>.+)=="
```
Comment (QuanNguyen-CDS) [inline tests/app/clients/test_aws_ses.py:103]: is there a reason why we are capturing name groups?

Comment (QuanNguyen-CDS) [inline tests/app/clients/test_aws_ses.py:103]: is there any worry about multiline regex compatibility across platforms? just a bit worried about this. No concrete example of it being wrong.

**Review by QuanNguyen-CDS — APPROVED** [review]
Comment are inline and minor concerns, not blocking the merge

File: tests/app/clients/test_aws_ses.py (hunk 1 above)
Comment (AntoineAugusti) [reply tests/app/clients/test_aws_ses.py:103]: Yes, those name groups are here to prove that envelopes have been set as expected. Some things need to be contained within things and groups are boundaries https://github.com/cds-snc/notification-api/blob/9ca52c3c0d44f745b4097b9a6d72cf854b6e3431/app/clients/email/aws_ses.py#L95-L106

Comment (AntoineAugusti) [reply tests/app/clients/test_aws_ses.py:103]: It passed at least on Linux and macOS, let's see if we get reports for Windows

**Pull request merged**